lib/src/main/kotlin/org/didcommx/didcomm/DIDComm.kt

```diff
@@ -148,21 +150,35 @@ class DIDComm(private val didDocResolver: DIDDocResolver, private val secretReso
         val (message, fromPriorIssuerKid) = packFromPrior(params.message, params.fromPriorIssuerKid, senderKeySelector)
         val (payload, signFromKid) = signIfNeeded(message.toString(), params, senderKeySelector)
         val (encryptedResult, recipientKeys) = encrypt(params, payload, senderKeySelector)
-        val (packedMessage) = protectSenderIfNeeded(params, encryptedResult, recipientKeys)
+        var (packedMessage) = protectSenderIfNeeded(params, encryptedResult, recipientKeys)
+
+        val didServicesChain = resolveDidServicesChain(
```

Comment:
Can we move `resolveDidServicesChain` to  `wrapInForwardIfNeeded` and return `serviceMetadata` as part of `wrapInForwardResult`, so that we encapsulate all forward-related logic there?

Reply:
Yes, but we have forward related business logic in packEncrypted (related to didServicesChain mostly). What if in the next version of forward protocol these chain will look and be processed differently? 
This is not critical, but I would make it as follows (both in Python and Kotlin): 
- from packEncrypted - pass forward input (headers, etc.) to a something from protocol/forward (probably internal function).
- that function returns the wrapped in forward message and some metadata (including serviceEndpoint). 
- packEncrypted uses these metadata to create the PackEncryptedMetadata